client/src/main/kotlin/io/spine/examples/chatspn/desktop/ChatPage.kt

```diff
@@ -0,0 +1,1038 @@
+    return leftPart + newMessage + rightPart
+}
+
+/**
+ * Creates the `MessageData` from the `MessageView`.
+ *
+ * @param client desktop client to find user profiles
+ */
+private fun MessageView.toMessageData(client: DesktopClient): MessageData {
+    return MessageData(
+        this.id,
+        client.findUser(this.user)!!,
+        this.content,
+        this.whenPosted
+    )
+}
+
+/**
+ * Creates the `MessageData` list. from the `MessageView` list.
+ *
+ * @param client desktop client to find user profiles
+ */
+private fun List<MessageView>.toMessageDataList(client: DesktopClient): MessageList {
+    val users = mutableMapOf<UserId, UserProfile>();
+    val messages = this.stream().map { message ->
+        val user: UserProfile
+        if (users.containsKey(message.user)) {
+            user = users[message.user]!!
+        } else {
+            user = client.findUser(message.user)!!
+            users[message.user] = user
+        }
+        MessageData(
+            message.id,
+            user,
+            message.content,
+            message.whenPosted
+        )
+    }.collect(Collectors.toList())
+    return messages
+}
+
+/**
+ * List of `MessageData`.
+ */
+public typealias MessageList = List<MessageData>
+
+/**
+ * Mutable state of messages in the chat.
+ */
+private typealias MutableMessagesState = MutableStateFlow<MessageList>
+
+/**
+ * Immutable state of messages in the chat.
+ */
+public typealias MessagesState = StateFlow<MessageList>
+
+/**
+ * Returns the new list without an element on provided index.
+ *
+ * @param index index of the element to remove
+ */
+private fun <T> List<T>.remove(index: Int): List<T> {
+    return this.subList(0, index) + this.subList(index + 1, this.size)
+}
+
+/**
+ * Defines whether the message at the provided index is the first in the message sequence
+ * from a particular user in the last 10 minutes.
+ */
+private fun MessageList.isFirstMemberMessage(index: Int): Boolean {
+    if (this.size - 1 < index) {
+        throw IndexOutOfBoundsException()
+    }
+    return index == 0 ||
+            !this[index].sender.equals(this[index - 1].sender) ||
+            (this[index].whenPosted.seconds - this[index - 1].whenPosted.seconds > 600)
+}
+
+/**
+ * Defines whether the message at the provided index is the last in the message sequence
+ * from a particular user in the last 10 minutes.
+ */
+private fun MessageList.isLastMemberMessage(index: Int): Boolean {
+    if (this.size - 1 < index) {
+        throw IndexOutOfBoundsException()
+    }
+    return index == this.size - 1 ||
+            !this[index].sender.equals(this[index + 1].sender) ||
+            (this[index + 1].whenPosted.seconds - this[index].whenPosted.seconds > 600)
+}
+
+/**
+ * Settings to display the single message.
+ */
+private data class MessageDisplaySettings(
+    val color: Color,
+    val shape: Shape,
+    val alignment: Alignment,
+    val arrowWidth: Float
+)
```

Comment:
This file is too long already. Let's split it by grouping related things by files.